Fladle 0.14.1 does not work with Gradle 7.0 Milestone 3

Environment: 
- Gradle 7.0 milestone 3
- Fladle 0.14.1

Relevant Fladle config in `:app` project:

```kts
fladle {
    serviceAccountCredentials.set(project.file(System.getenv("GCLOUD_CREDENTIALS") ?: "${System.getenv("HOME")}/.config/gcloud/application_default_credentials.json"))
    // Optional parameters
    useOrchestrator.set(false)
    environmentVariables.set(mapOf(
        "clearPackageData" to "true",
        "coverage" to "true",
        "coverageFile" to "/sdcard/coverage.ec"
        // coverageFile ONLY used when Orchestrator is off.
        // otherwise, you need to use `coverageFilePath`, each Orchestrator will make a separate coverage.ec
    ))
    debugApk.set("$buildDir/outputs/apk/beta/debug/app-beta-debug.apk")
    instrumentationApk.set("$buildDir/outputs/apk/androidTest/beta/debug/app-beta-debug-androidTest.apk")
    testTimeout.set("30m")
    recordVideo.set(false)
    performanceMetrics.set(false)
    variant.set("beta")
    directoriesToPull.set(listOf("/sdcard/"))
    filesToDownload.set(listOf(
        "test_result_*.xml",
        // regex, not glob
        ".*/sdcard/[^/]+.ec\$" // coverage files.
    ))
    // see the device list like this:
    // $ gcloud firebase test android models list
    configs {
        create("emulators") {
            testShards.set(5)
            devices.set(listOf(
                // emulators:
                mapOf("model" to "NexusLowRes", "version" to "25", "orientation" to "portrait", "language" to "en")
            ))
        }
        create("physical") {
            devices.set(listOf(
                mapOf("model" to "sailfish", "version" to "25", "orientation" to "portrait", "language" to "en") // Pixel (1)
            ))
        }
    }
    shardTime.set(300) // seconds
    testShards.set(5)
}
```

Gradle errors with this:


```
> Task :app:writeConfigPropsEmulators FAILED

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':app:writeConfigPropsEmulators' (type 'YamlConfigWriterTask').
  - Type 'YamlConfigWriterTask' property 'config.name$fladle' is missing an input or output annotation.
     
    Reason: A property without annotation isn't considered during up-to-date checking.
      
      Possible solutions:
        1. Add an input or output annotation.
        2. Mark it as @Internal.
      
      Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#missing_annotation for more details about this problem.
```

--- a/fladle-plugin/src/main/java/com/osacky/flank/gradle/FladleConfigImpl.kt
+++ b/fladle-plugin/src/main/java/com/osacky/flank/gradle/FladleConfigImpl.kt
@@ -4,9 +4,10 @@ import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.provider.ListProperty
 import org.gradle.api.provider.MapProperty
 import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Internal
 
 data class FladleConfigImpl(
-  internal val name: String,
+  @get:Internal internal val name: String,
   override val projectId: Property<String>,
   override val serviceAccountCredentials: RegularFileProperty,
   override val debugApk: Property<String>,

--- a/fladle-plugin/src/test/java/com/osacky/flank/gradle/integration/FlankGradlePluginIntegrationTest.kt
+++ b/fladle-plugin/src/test/java/com/osacky/flank/gradle/integration/FlankGradlePluginIntegrationTest.kt
@@ -3,6 +3,7 @@ package com.osacky.flank.gradle.integration
 import com.google.common.truth.Truth.assertThat
 import org.gradle.testkit.runner.BuildResult
 import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
 import org.junit.Rule
 import org.junit.Test
 import org.junit.rules.TemporaryFolder
@@ -212,7 +213,7 @@ class FlankGradlePluginIntegrationTest {
   }
 
   @Test
-  fun testSpecifyingBothInstrumenationAndRoboscriptFailsBuild() {
+  fun testSpecifyingBothInstrumentationAndRoboscriptFailsBuild() {
     writeBuildGradle(
       """plugins {
             id "com.osacky.fladle"
@@ -220,7 +221,7 @@ class FlankGradlePluginIntegrationTest {
            fladle {
              serviceAccountCredentials = project.layout.projectDirectory.file("foo")
              debugApk = "test-debug.apk"
-             instrumentationApk = "instrumenation-debug.apk"
+             instrumentationApk = "instrumentation-debug.apk"
              roboScript = "foo.script"
            }
       """.trimIndent()
@@ -234,5 +235,32 @@ class FlankGradlePluginIntegrationTest {
       .buildAndFail()
 
     assertThat(result.output).contains("Both instrumentationApk file and roboScript file were specified, but only one is expected.")
+  }
+
+  @Test
+  fun testGradleSevenCompat() {
+    writeBuildGradle(
+      """plugins {
+           id "com.osacky.fladle"
+         }
+         fladle {
+           serviceAccountCredentials = project.layout.projectDirectory.file("foo")
+           debugApk = "test-debug.apk"
+           instrumentationApk = "instrumentation-debug.apk"
+           configs {
+             fooConfig {
+             }
+           }
+         }
+         """.trimMargin()
+    )
+    testProjectRoot.newFile("foo").writeText("{}")
+    val result = GradleRunner.create()
+      .withProjectDir(testProjectRoot.root)
+      .withPluginClasspath()
+      .withGradleVersion("7.0-milestone-3")
+      .withArguments("printYmlFooConfig")
+      .build()
+    assertThat(result.task(":printYmlFooConfig")!!.outcome).isEqualTo(TaskOutcome.SUCCESS)
   }
 }

